### PR TITLE
Add ssh key and password generation

### PR DESCRIPTION
**Pull request by jagee**
Signed-off-by: Jeremy Agee <jeremy.agee@gmail.com>

### CODE REVIEW

File: contrib/make-cloud-init-iso.sh [hunk 1]
```diff
@@ -1,39 +1,152 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-PUB=${PUB-~/.ssh/id_rsa.pub}
-USER=${USER-cloudrouter}
-HOST=${HOST-localhost}
+if [ ! -n "${HOST}" ]; then
+    HOST="localhost"
```
Comment (abn) [inline contrib/make-cloud-init-iso.sh]: Any reason why this changed? Same behaviour no?

Comment (jagee) [inline contrib/make-cloud-init-iso.sh]: No big reason, i could use the other method as well.

Comment (abn) [inline contrib/make-cloud-init-iso.sh]: just curious :)

File: contrib/make-cloud-init-iso.sh [hunk 2]
```diff
@@ -1,39 +1,152 @@
...
-if [ ! -f "${PUB}" ]; then
-    echo >&2 "[ERROR] Cloud not find pub key file at ${PUB}"
+if [ ! -n "${PRIKEY}" ] && [ ! -n "${PUBKEY}" ]; then
+    PRIKEY="${WORKING_DIR}/cloudrouter_rsa"
+    PUBKEY="${PRIKEY}.pub"
 fi
 
-type genisoimage > /dev/null 2>&1 \
-    || { echo >&2 "[ERROR] genisoimage command not found."; exit 1; }
+for cmd in "genisoimage ssh-keygen"; do
```
Comment (abn) [inline contrib/make-cloud-init-iso.sh]: no quotes here

**Comment by jagee** [comment]
ok fixed the quotes here, doh

**Pull request merged**